src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1954,14 +1957,19 @@ void ProcessServerVersion(string value)
             }
             if (!versionString.Contains('.'))
                 versionString += ".0";
-            ServerVersion = new Version(versionString);
+            return versionString;
         }
 
         /// <summary>
         /// Whether the backend is an AWS Redshift instance
         /// </summary>
         bool IsRedshift => Settings.ServerCompatibilityMode == ServerCompatibilityMode.Redshift;
 
+        /// <summary>
+        /// Whether the backend is a CrateDB instance
+        /// </summary>
+        internal bool IsCrateDB => CrateDBVersion >= new Version(2, 3, 0);
```

Comment:
I'd rather you threw an informative exception if trying to connect to an unsupported version of CrateDB.